Source/WebCore/platform/network/NetworkStorageSession.cpp

```diff
@@ -210,6 +210,17 @@ String NetworkStorageSession::cookiePartitionIdentifier(const ResourceRequest& r
     return cookiePartitionIdentifier(request.firstPartyForCookies());
 }
 
+String NetworkStorageSession::cookiePartitionIdentifier(const SecurityOriginData& origin)
+{
+    ASSERT(std::holds_alternative<SecurityOriginData::Tuple>(origin.data()));
+
+    return switchOn(origin.data(), [&origin](const SecurityOriginData::Tuple&) {
+        return cookiePartitionIdentifier(origin.toURL());
+    }, [](const ProcessQualified<OpaqueOriginIdentifier>&) {
+        return String { };
+    });
```

Comment:
Opaque origin gets empty url so you could just return `cookiePartitionIdentifier(origin.toURL())` (and this function might be unnecessary if you could do `origin.toURL()` in the caller).

Reply:
I'm not sure we should ever have an opaque origin in this case, but that is a good point.